src/averageAges.js

```diff
@@ -55,7 +75,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children;
+  let women;
+
+  (!onlyWithSon)
+
+    ? children = people.filter(person =>
+      people.find(woman =>
+        woman.name === person.mother))
+
+    : children = people.filter(person =>
+      people.find(woman =>
+        woman.name === person.mother) && person.sex === 'm');
+
+  (!onlyWithSon)
+
+    ? women = people.filter(person => people.find(child =>
+      child.mother === person.name))
+
+    : women = people.filter(person => people.find(child =>
+      child.mother === person.name
+      && child.sex === 'm'));
+
+  const agesDiff = children.map(child =>
+    (child.born - women.find(woman =>
+      child.mother === woman.name).born));
```

Comment:
I'm not sure, but can we use here people array instead of women and delete the women variable at all? It will be less code and -1 loop )

```suggestion
  const agesDiff = children.map(child =>
    (child.born - people.find(woman =>
      child.mother === woman.name).born));
```